### PR TITLE
Upgrade to v2.36.0 of the AzureRM Terraform Provider

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #762

### CODE REVIEW

File: sdk/dotnet/Network/Inputs/VpnGatewayBgpSettingsArgs.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@
 namespace Pulumi.Azure.Network.Inputs
 {
 
-    public sealed class VpnGatewayBgpSettingArgs : Pulumi.ResourceArgs
+    public sealed class VpnGatewayBgpSettingsArgs : Pulumi.ResourceArgs
```
Comment (mikhailshilkov) [inline sdk/dotnet/Network/Inputs/VpnGatewayBgpSettingsArgs.cs:13]: What happened to this name? It's a breaking change.

File: sdk/dotnet/Network/NetworkConnectionMonitor.cs [hunk 2]
```diff
@@ -9,121 +9,53 @@
...
         public Output<string> Name { get; private set; } = null!;
 
         /// <summary>
-        /// The name of the Network Watcher. Changing this forces a new Network Connection Monitor to be created.
+        /// The ID of the Network Watcher. Changing this forces a new resource to be created.
         /// </summary>
-        [Output("networkWatcherName")]
-        public Output<string> NetworkWatcherName { get; private set; } = null!;
+        [Output("networkWatcherId")]
+        public Output<string> NetworkWatcherId { get; private set; } = null!;
```
Comment (mikhailshilkov) [inline sdk/dotnet/Network/NetworkConnectionMonitor.cs:45]: This bunch of changes is breaking too

File: sdk/dotnet/Network/VpnGateway.cs [hunk 3]
```diff
@@ -64,7 +64,7 @@ public partial class VpnGateway : Pulumi.CustomResource
         /// A `bgp_settings` block as defined below.
         /// </summary>
         [Output("bgpSettings")]
-        public Output<ImmutableArray<Outputs.VpnGatewayBgpSetting>> BgpSettings { get; private set; } = null!;
+        public Output<Outputs.VpnGatewayBgpSettings> BgpSettings { get; private set; } = null!;
```
Comment (mikhailshilkov) [inline sdk/dotnet/Network/VpnGateway.cs:67]: Aha, that's what happened: an array replaced with a type

File: sdk/dotnet/Network/Inputs/VpnGatewayBgpSettingsArgs.cs (hunk 1 above)
Comment (stack72) [reply sdk/dotnet/Network/Inputs/VpnGatewayBgpSettingsArgs.cs:13]: They *added* MaxItems: 1` here :/ https://github.com/terraform-providers/terraform-provider-azurerm/pull/9035/files#diff-a755240392b1658fa6f3ba7a1af73a46b17e2c07139cd248fd161195ac2ff8ecR64

File: sdk/dotnet/Network/NetworkConnectionMonitor.cs (hunk 2 above)
Comment (stack72) [reply sdk/dotnet/Network/NetworkConnectionMonitor.cs:45]: This is a flat out breaking change in their SDK... 

https://github.com/terraform-providers/terraform-provider-azurerm/pull/8640/files#diff-746c9ec7da44dc1d034172a67bfb5a97081472100c3c1eeb89f0e26e8fa704acR50

**Comment by stack72** [comment]
As per internal discussions, we are going to "accept" these breaking changes in a minor release and add details to the CHANGELOG about them for users

**Review by mikhailshilkov — APPROVED** [review]

**Pull request merged**